src/tools/compiletest/src/runtest.rs

```diff
@@ -2272,23 +2272,15 @@ impl<'test> TestCx<'test> {
 
         let mut tested = 0;
         for _ in res.stdout.split('\n').filter(|s| s.starts_with("test ")).inspect(|s| {
-            if let Some((left, right)) = s.split_once(" - ") {
-                let path = left.rsplit("test ").next().unwrap();
+            if let Some((left, _)) = s.split_once(" - ") {
+                let path_with_line = left.rsplit("test ").next().unwrap();
+                let (path, line_number) =
+                    path_with_line.rsplit_once(":").expect("line number not found");
                 let path = fs::canonicalize(&path).expect("failed to canonicalize");
                 let path = path.to_str().unwrap().replace('\\', "/");
                 if let Some(ref mut v) = files.get_mut(&path) {
                     tested += 1;
-                    let mut iter = right.split("(line ");
-                    iter.next();
-                    let line = iter
-                        .next()
-                        .unwrap_or(")")
-                        .split(')')
-                        .next()
-                        .unwrap_or("0")
-                        .parse()
-                        .unwrap_or(0);
-                    if let Ok(pos) = v.binary_search(&line) {
+                    if let Ok(pos) = v.binary_search(&line_number.parse::<usize>().unwrap_or(0)) {
```

Comment:
Discussion: hm, this seems rather fragile 🤔 But I don't feel like blocking on this.